core/mondoo-kubernetes-security.mql.yaml

```diff
@@ -3037,19 +2847,8 @@ queries:
     title: Container should not run as root
     impact: 100
     mql: |
-      k8s.deployment {
-        podSecurityContext=podSpec['securityContext']
-        initContainers {
-          a = podSecurityContext['runAsNonRoot'] == true && securityContext['runAsNonRoot'] == null
-          res = securityContext['runAsNonRoot'] == true || a
-          res == true
-        }
-        containers {
-          a = podSecurityContext['runAsNonRoot'] == true && securityContext['runAsNonRoot'] == null
-          res = securityContext['runAsNonRoot'] == true || a
-          res == true
-        }
-      }
+      k8s.deployment.containers.all( securityContext['runAsNonRoot'] == true )
```

Comment:
Does the `k8s.deployment.` check also include this? (still need to double-check myself)

```
        It is also possible to set it for all containers at the Pod level:

        ```yaml
        ---
        apiVersion: v1
        kind: Pod
        spec:
          securityContext:
            runAsNonRoot: true
```